### ISSUE TITLE
[restrict-plus-operands] rule doesn't support bigint

### ISSUE THREAD

**Issue by novemberborn** [typescript-eslint/typescript-eslint#309]
**Repro**

```JSON
{
  "plugins": ["@typescript-eslint"],
  "rules": {
    "@typescript-eslint/restrict-plus-operands": "error"
  }
}
```

```TS
1n + 1n
```

**Expected Result**

No errors.

**Actual Result**

The operands are not numbers, but they're `bigint`s which are still addable without type conversion.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.4.0` |
| `@typescript-eslint/parser`        | `1.4.0` |
| `TypeScript`                       | `3.2.1` |
| `ESLint`                           | `X.Y.Z` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by webschik** [comment]
Hi!

I've created a PR with the fix #310.

**Comment by novemberborn** [comment]
This fixes `1n + 2n` but not `BigInt(1) + 2n` or `const foo = 1n; foo + 2n`.

"bigint" should be mentioned in the error messages. (Yes I can PR that one. Not sure about the type checking for the other use cases.)
